### PR TITLE
DEV: bots are always allowed to chat

### PR DESCRIPTION
**Pull request by jjaffeux**
Bots should be allowed to chat regardless of their groups, just like staff. It makes configuring bots to work in chat much easier.

### CODE REVIEW

**Review by SamSaffron — APPROVED** [review]

File: plugins/chat/lib/chat/guardian_extensions.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def can_moderate_chat?(chatable)
 
     def can_chat?
       return false if anonymous?
-      @user.staff? || @user.in_any_groups?(Chat.allowed_group_ids)
+      is_staff? || @user.bot? || @user.in_any_groups?(Chat.allowed_group_ids)
```
Comment (jjaffeux) [inline plugins/chat/lib/chat/guardian_extensions.rb:16]: the rename from `@user.staff?` to `is_staff?` is not related to the change, just felt more consistent

Comment (jjaffeux) [inline plugins/chat/lib/chat/guardian_extensions.rb:16]: Adding `@user.bot?` is the core of the change

**Comment by SamSaffron** [comment]
approved, but a test to document the bot behavior would be nice, its only a few lines.

**Comment by jjaffeux** [comment]
> approved, but a test to document the bot behavior would be nice, its only a few lines.

Sorry I missed that comment, I did add a test to document the change in the guardian method. What other test were you thinking about? Like some kind of end to end test where a bot can post a message in a channel even if he is not in the allowed groups? I can make this yes.

**Pull request merged**